### PR TITLE
Updated: Implement Symbol.prototype.description

### LINKED ISSUE #962: Implement ES2019 Symbol.prototype.description
Proposal: https://github.com/tc39/proposal-Symbol-description
Spec: https://262.ecma-international.org/12.0/#sec-symbol.prototype.description

### PR DESCRIPTION
**Pull request by gbrail**
This includes the changes made by @andreabergia to:

* Add Symbol.prototype.description
* Move Symbol to use lambdas

Since so many things have changed in this area recently, and since the lambda
implementation wasn't totally ready for this, and since I wanted this to 
use Lambdas appropriately AND move forward, this PR includes @andreabergia's
original commits and a few more tweaks on top.

Closes #962

### CODE REVIEW

File: tests/testsrc/test262.properties [hunk 1]
```diff
@@ -2312,25 +2312,15 @@ built-ins/Symbol 36/94 (38.3%)
     asyncDispose/prop-desc.js
     asyncIterator/prop-desc.js
     dispose/prop-desc.js
-    for/cross-realm.js
```
Comment (rbri) [inline tests/testsrc/test262.properties:2315]: why does the overall count does not change? Have you regenerated the file or just removed the failing tests?

**Comment by rbri** [comment]
+1

**Comment by gbrail** [comment]
Yes, I updated test262.properties manually.

I just re-ran the tests with "updateTest262properties=true" and it produced 8000 lines of changes. Maybe I'm doing something wrong? I don't know how to begin to verify whether that's correct.

**Comment by rbri** [comment]
Will try it here.... give me 5 min

**Comment by rbri** [comment]
use this and you are fine

built-ins/Symbol 26/94 (27.66%)

**Comment by rbri** [comment]
seems like the head version of the file is already a bit out of sync - will make a pr

**Comment by rbri** [comment]
@gbrail test262.properties is updated on head, simple fix by replacing the appropriate line with 'built-ins/Symbol 26/94 (27.66%)' and you are done and ready to merge

**Comment by gbrail** [comment]
Thanks! I need to work on my real job for a bit but I'll get to it tonight!

**Comment by rbri** [comment]
> Thanks! I need to work on my real job for a bit but I'll get to it tonight!

Have fun with both....

**Comment by rbri** [comment]
Thanks @andreabergia and @gbrail!

**Pull request merged**